### ISSUE TITLE
AuthStrategy Documentation

### ISSUE THREAD

**Issue by avalenti89** [pedroslopez/whatsapp-web.js#1258]
With the new AuthStrategies' release (1.16.*), the documentation must be updated with the same example made for old LegacyAuth, because there is no example about how to get QR code(or any other session identifier) or how to check if a session is valid or not.
Right now I'm forced to use LegacyAuth

**Comment by tonvitalferreira** [comment]
I need this to! ;)

**Comment by pedropuppim** [comment]
![image](https://user-images.githubusercontent.com/774154/155995292-971a5809-3ae9-468f-8852-b001fc337d1d.png)

Using this code, it already saves the session with the identifier "client-one" in the .wwebjs_auth folder.

**Comment by ItHustleCoder** [comment]
> With the new AuthStrategies' release (1.16.*), the documentation must be updated with the same example made for old LegacyAuth, because there is no example about how to get QR code(or any other session identifier) or how to check if a session is valid or not. Right now I'm forced to use LegacyAuth

Here the documentation 
https://wwebjs.dev/guide/authentication.html#restoring-the-session

**Comment by avalenti89** [comment]
> Here the documentation
https://wwebjs.dev/guide/authentication.html#restoring-the-session

The documentation is written for LegacySessionAuth, where the session is controlled by hand.
There is no documentation about this for new LocalAuth.

**Comment by avalenti89** [comment]
> ![image](https://user-images.githubusercontent.com/774154/155995292-971a5809-3ae9-468f-8852-b001fc337d1d.png)
> 
> Using this code, it already saves the session with the identifier "client-one" in the .wwebjs_auth folder.

yes, but I need a file (or another way) to get the session information to generate a QR.
I think that the Client must have a local state or a way to query the session

**Comment by MorpheusFT** [comment]
I need this too.

Can't figure out how to restore a session when using LocalAuth.

LegacyAuth stopped working for me this morning after whatsapp updated (multi device forced?)

**Comment by FilmonK** [comment]
I've been testing this with decent results and will keep testing, but this has worked for me as I was in the same situation. And Legacy was completely broken for me and wasn't an alternative.

<img width="1071" alt="Screen Shot 2022-03-01 at 4 26 56 PM" src="https://user-images.githubusercontent.com/17398657/156251875-71912c28-4e6b-487f-ac6d-d4773009e1c6.png">

**Comment by dexv** [comment]
Hi. For version 1.12.5:

![image](https://user-images.githubusercontent.com/64558868/156596112-dc327fab-56c0-4e08-9db3-fd63feea5da2.png)

**Comment by rien333** [comment]
Please don't use images to show code. Makes it harder for others to locate issues, and makes it impossible to copy/paste. If you want syntax highlighting, github has a method for that.

**Comment by pedroslopez** [comment]
@avalenti89 LocalAuth works by providing a [user data directory](https://chromium.googlesource.com/chromium/src/+/master/docs/user_data_dir.md) to the browser, which persists the data the same way it would if you were using WhatsApp Web through your personal browser. As such, what's actually in this directory is opaque and its contents should not matter as it's controlled by the browser itself.

The reason why LocalAuth does not have documentation for getting session info is because it does not provide it nor should need to.

**Comment by jalescardoso** [comment]
I'm on version 1.16.3 and I can't restore a session using LocalAuth

**Comment by borangboring** [comment]
Me too, I also used localAuth() but everytime I ran the program it just came back to qr-code scanning.

When I went into the .wwebjs_auth. There's only an empty folder titled "session". Nothing else.

Now, I think it's platform specific issue, because I used the exact same code and dependencies in Windows and it worked perfectly. The filesystem & authorization had no issue.

The problem raised when I use Termux (an Android app) with proot Linux (Ubuntu Focal fossa)
